samples/javascript/scoreboard/src/server/app.ts

```diff
@@ -0,0 +1,83 @@
+import express from 'express'
+import { WebPubSubServiceClient } from '@azure/web-pubsub'
+import { UserEventRequest, UserEventResponseHandler, WebPubSubEventHandler } from '@azure/web-pubsub-express'
+import constants from '../common/constants'
+import matchGenerator from './matchData/matchGenerator'
+import { MatchSummaryListPayload } from '../models/payloads/MatchSummaryListPayload'
+import { MatchRunner } from './matchData/matchRunner'
+import utils from '../common/utils'
+import { MatchTeams } from '../models/MatchTeams'
+import path from 'path'
+
+const appRoot = require('app-root-path')
+
+// environment
+const port = process.env.port || process.env.PORT || 5050
+const staticRoot = path.join(appRoot.path,(process.env.NODE_ENV == 'production' ? 'dist/dist-client' : 'public'))
+const connectionString = process.env.CONN_STR as string
+const hubName = process.env.NODE_ENV === 'production' ? 'scoreboard' : 'dev_scoreboard'
+
+const app = express()
+
+const serviceClient = new WebPubSubServiceClient(connectionString, hubName, { allowInsecureConnection: true })
+const handler = new WebPubSubEventHandler(hubName, {
+    path: '/eventhandler',
+    onConnected: async req => {
+        console.log(`${req.context.userId} connected`)
+    },
+    handleUserEvent: async (req: UserEventRequest, res: UserEventResponseHandler) => {
+        const user = req.context.userId as string
+        switch (req.context.eventName) {
+            // reply live match list for the user event 'getLiveMatchList'
+            case constants.eventNames.getLiveMatchList:
+                await serviceClient.sendToUser(user, <MatchSummaryListPayload>{
+                    event: constants.eventNames.getLiveMatchList,
+                    list: matchGenerator.liveMatchList.map(m => m.getSummary()),
+                })
+                break
+            // reply past match list for the user event 'getPastMatchList'
+            case constants.eventNames.getPastMatchList:
+                await serviceClient.sendToUser(user, <MatchSummaryListPayload>{
+                    event: constants.eventNames.getPastMatchList,
+                    list: matchGenerator.pastMatchList.map(m => m.getSummary()),
+                })
+                break
+            case constants.eventNames.realtimeMatchDetails: {
+                const teams: MatchTeams = req.data as any
+                const liveMatches = matchGenerator.liveMatchList.filter(m => utils.getId(m.teams) === utils.getId(teams))
```

Comment:
Is it makes more sense to always use `matchRunner` to get live data or history data? the `matchGenerator` is one way to get the match data.